public/service-worker.js

```diff
@@ -277,10 +278,15 @@ self.addEventListener("fetch", (event) => {
 
     // Tries to serve the requested data from the DATA_CACHE
     // If unable to, then it makes a network fetch and stores the fetched data in the cache
+
+    // ML: Why would this third branch ever be taken? why would going to /bookmarks_in_pipeline not be in "navigate mode"
   } else if (requestUrl.pathname.startsWith("/bookmarks_in_pipeline")) {
     event.respondWith(
       caches.open(DATA_CACHE).then((cache) =>
         cache.match(event.request, { ignoreSearch: true }).then((response) => {
+          // ML: as far as we can tell, once the words have been cached, they will never be refreshed
```

Comment:
This is also related to my response about the function that deletes caches. 
At service worker activation if the cache key does not match the current version, then the outdated cache is deleted and the service worker fetch and stores the update data.